h2o-py/h2o/h2o.py

```diff
@@ -288,7 +288,7 @@ def get_mem_size(mmint, mmgb):
                                   min_mem_size=mmin, ice_root=ice_root, log_dir=log_dir, log_level=log_level,
                                   max_log_file_size=max_log_file_size, port=port, name=name,
                                   extra_classpath=extra_classpath, jvm_custom_args=jvm_custom_args,
-                                  bind_to_localhost=bind_to_localhost)
+                                  bind_to_localhost=bind_to_localhost, off_heap_memory_ratio=kwargs.get("off_heap_memory_ratio", 2.0/3))
```

Comment:
> :param off_heap_memory_ratio: Minimum amount of memory that should be unused by the JVM.
> "General recommendation is to keep 1/3 of available memory for XGBoost. "

and here we set a default value = 2/3  ???

Reply:
🤦 Good catch, thank you! I must have confused myself by reading the git message which contained the example setting the value to 0.75 and then reading the docs saying the limit should be 2/3.

Do I understand correctly that you are proposing something like calling `h2o.rapids("(setproperty \"sys.ai.h2o.xgboost_check_memory_ratio\" \"0.666\")")` within `h2o.init` after it connects (to make sure that we have the property set just for those kinds of setup)? 

I don't like that since I didn't make up the `off_heap_memory_ratio` and since it's already existing we would have two ways of doing almost the same thing. The `off_heap_memory_ratio` is now responsible for logging a message about XGBoost's potential memory issues during the startup so by using the `setproperty` rapid I might introduce race condition.

Reply:
> Do I understand correctly that you are proposing something like calling h2o.rapids("(setproperty \"sys.ai.h2o.xgboost_check_memory_ratio\" \"0.666\")") within h2o.init after it connects (to make sure that we have the property set just for those kinds of setup)?

nope, I definitely don't suggest anything like that.

First, i think the requirements in the ticket are weird. As i raised in an old comment: if it's only about raising warnings about backend memory, then I can try to look at it from all angles, I still don't understand why it should be activated only on client start.

Second, I find it very convoluted to create params that become pointless as soon as you know their existence: it's basically a `warn_me_later_if_i_forgot_to_set_memory_size_after_this_param` param, it's sooo weird.
We could add hundreds of init params like this one that each just enable a single warning, and we would all agree this would be nonsense.
System properties are much less invasive, so they don't have this issue: you can create as many as you want without confusing the user.
Also, system properties can be set for a certain scope only: it's easy to imagine enforcing the 2/3-1/3 rule only for an AutoML run, as AutoML w/ XGB seems to be our main concern here.

>  I don't like that since I didn't make up the off_heap_memory_ratio 

I just don't think anymore it was a good idea to expose this as an `H2O.ARGS` in the first place given that it's not functional: again, it's an extremely complex way to just switch a warning. As it is not exposed, it's definitely not used by anyone (current default value = 0) and could be removed from those args.

> The off_heap_memory_ratio is now responsible for logging a message about XGBoost's potential memory issues during the startup so by using the setproperty rapid I might introduce race condition.

what happens at startup uses the value of System.property at startup.
You added a warning that is checked again for each XGB builder, each of those would use the value of the system property at that time.

What I would suggest is to add the logic in `AutoML.java` itself, when validating params.
There, we know if XGB will be used or not, we can override the sys property value only if it is currently unset, and we can restore it after the automl run.
And I don't see why we should do that only if backend was started from Py/R... but even so, we could apply the logic above only in single node then.